language.md

```diff
@@ -475,6 +469,37 @@ Performs multiple collections at once and converts the results into an object.
 This can be very useful to share a `Where` condition while collecting multiple
 pieces of information.
 
+- [_behaviour_] `` ` `` _collector_ `` ` ``
+
+This allows using optional values in other collectors. For instance, suppose
+there is an optional number and the minimum is desired, one could write: `` `
+Min x? ` ``.
+
+There are special _behaviours_ for how to handle records with missing data:
+
+- the unspecified behaviour is to simply drop empty optionals and proceed with
+  the collector as normal. This is equivalent to ``Where x != ` ` Min x Default
+  1234 ``; since `x` will never have the empty optional, the default will never
+  be used.
+- `OnlyIf All` requires that no empty input makes it to the collector; if any
+  optional input is found, the output from the collector is replaced by the
+  empty optional. So, if ``v = OnlyIf All `Min x?` `` was given `` `3` and `` ` `
+  ``, it would produce `` v == ` ` ``.
+- `OnlyIf Any` requires that one non-empty input makes it to the collector; if
+  no optional input is found, the output from the collector is replaced by the
+  empty optional. So, if ``v = OnlyIf Any `Min x?` `` was given `` `3` and
+  `` ` ` ``, it would produce `` v == `3` ``, but `` ` ` `` and `` ` ` `` would
+  produce ``v == ` ` ``.
+- `Require All` requires that no empty input makes it to the collector; if any
+  optional input is found, the group will be rejected. So, if
+  ``v = Require All `Min x?` `` was given `` `3` and `` ` ` ``, it would not be
+  present in the output.
+- `Require Any` requires that one non-empty empty input makes it to the
+  collector; if no optional input is found, the group will be rejected. So, if
+  ``v = Require Any `Min x?` `` was given `` `3` and `` ` ` ``, it would produce
+  `v == 3`, but `` ` ` `` and `` ` ` `` would not be present in the output.
```

Comment:
why is `  ` twice here?

Reply:
I guess it doesn't have to be. I thought it was clearer to have matched numbers of inputs and outputs.

Reply:
but it is only once in the input?

Reply:
No, it could be multiple times in the input. Any number of empties would be treated a problem.

Reply:
I feel like i'm missing something. is this a rendering problem? there is one blank after 'was given' but two blanks in the example of what wouldn't be in the output 
![image](https://github.com/oicr-gsi/shesmu/assets/17747491/ff316014-89a8-4bae-9e57-7f959dec74a7)

Reply:
Oh, yeah, that's not how Jekyll renders it. Sigh. It's supposed to be like this ignoring linebreaks:

 ```
v = Require Any `Min x?`
```
 was given
```
 `3`
```
 and
```
 ` `
```
, it would produce
```
v == 3
```
, but
```
 ` `
```
 and
```
` `
```
would not be present in the output.

Reply:
Yeah, that's what the document says. Why is there 1 blank in the first clause and 2 blanks in the second?
I would think it would either be 'if v = Require Any Min x? was given 3, ' ', and ' ', it would produce 'v == 3' but ' ' and ' ' would not be present in the output'
or
'if v = Require Any Min x? was given 3 and ' ', it would produce 'v == 3' but ' ' would not be present in the output'

Reply:
I think I fixed the formatting. The second one is "only of blanks".